presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -358,8 +358,12 @@ PlanNodeStatsEstimate addJoinComplementStats(
             PlanNodeStatsEstimate innerJoinStats,
             PlanNodeStatsEstimate joinComplementStats)
     {
-        double innerJoinRowCount = innerJoinStats.getOutputRowCount();
         double joinComplementRowCount = joinComplementStats.getOutputRowCount();
+        if (joinComplementRowCount == 0) {
+            return innerJoinStats;
```

Comment:
Can we have a test for that? Especially in one of the `TestTpc*Stats` classes?

Reply:
`TestTpchLocalStats#testFullJoin`, `TestTpchLocalStats#testRightJoin`, `TestTpchLocalStats#testLeftJoin` tests cover this branch

Reply:
as in https://github.com/prestodb/presto/pull/11895#discussion_r233997741

Reply:
There is no division `joinComplementRowCount` anywhere. Should the check be: `outputRowCount == 0` instead?

Reply:
I had that like this before. But in the previous PR you suggested to shortcircut if complement stats are 0. Correct me if i got you wrong.

Reply:
It is not in the loop. It is at the beginning of the `addJoinComplementStats`.